enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/net/Server.java

```diff
@@ -120,14 +143,54 @@ public synchronized void stop()
         catch ( InterruptedException e )
         {
             Thread.currentThread().interrupt();
-            dbgLog.warn( "Interrupted while closing channel." );
+            debugLog.warn( "Interrupted while closing channel." );
         }
 
         if ( workerGroup != null &&
                 workerGroup.shutdownGracefully( 2, 5, TimeUnit.SECONDS ).awaitUninterruptibly( 10, TimeUnit.SECONDS ) )
         {
-            dbgLog.warn( "Worker group not shutdown within 10 seconds." );
+            debugLog.warn( "Worker group not shutdown within 10 seconds." );
         }
         workerGroup = null;
     }
+
+    public ListenSocketAddress address()
+    {
+        return listenAddress;
+    }
+
+    public int childCount()
+    {
+        return childCount.get();
+    }
+
+    private class ChildCounter extends ChannelInboundHandlerAdapter
+    {
+        private final ChannelInitializer<SocketChannel> childInitializer;
+
+        ChildCounter( ChannelInitializer<SocketChannel> childInitializer )
+        {
+            this.childInitializer = childInitializer;
+        }
+
+        @Override
+        public void channelActive( ChannelHandlerContext ctx ) throws Exception
+        {
+            childCount.incrementAndGet();
```

Comment:
Capturing `childCount` from the outer class is perhaps a little confusing.

Reply:
I moved the child counting functionality to the handler injected by the test instead.